lib/usdUfe/utils/editRouter.cpp

```diff
@@ -229,14 +219,21 @@ getEditRouterEditTarget(const PXR_NS::TfToken& operation, const PXR_NS::UsdPrim&
 
     const auto found = routingData.find(EditRoutingTokens->EditTarget);
 
-    if (found != routingData.end()) {
-        const auto& value = found->second;
-        if (value.IsHolding<PXR_NS::UsdEditTarget>()) {
-            const auto editTarget = value.Get<PXR_NS::UsdEditTarget>();
-            return editTarget;
-        }
+    if (found == routingData.end())
+        return PXR_NS::UsdEditTarget();
+
+    const auto& value = found->second;
+
+    if (value.IsHolding<PXR_NS::UsdEditTarget>()) {
+        auto&& editTarget = value.Get<PXR_NS::UsdEditTarget>();
+        return editTarget;
     }
-    return PXR_NS::UsdEditTarget();
+
+    auto layer = _extractLayer(found->second);
+    if (!layer)
+        PXR_NS::UsdEditTarget();
+
+    return PXR_NS::UsdEditTarget(layer);
```

Comment:
So essentially this allows users use pass in a SdfLayer and get an UsdEditTarget back, right? That's nice to have for local layer use cases.

Reply:
Yes, the other routable commands allow the routing function to return a layer or layer ID, so this makes delete support those too.

We would still need to update the other routable commands to use your trick to allow a non-local layers, but that would be done in a future pull request.

Reply:
Sounds good.